pkg/client/restclient/request.go

```diff
@@ -105,16 +106,13 @@ type Request struct {
 	resource     string
 	resourceName string
 	subresource  string
-	selector     labels.Selector
 	timeout      time.Duration
 
 	// output
 	err  error
 	body io.Reader
 
-	// The constructed request and the response
-	req  *http.Request
-	resp *http.Response
+	ctx context.Context
```

Comment:
Are you planning to use this for anything other than the cancel func?  If not, is there significant benefit to allowing control of the full context?  It seems like letting caller set a `requestTimeout` (yes, I agree its weird with that query param thing above), will be easier API-wise.